strapi/strapi
# fix: Remove Private Attributes from GraphQL Filters and Inputs

### What does it do?

Prevent attributes tagged as private to appear in the GraphQL Input and Filters schema definitions.

TODO:

- [x] u&p registration for graphql should manually add password as an exception to prevent a breaking change and to mirror content API register behaviour
- [x] add test that private fields aren't visible in schema anymore

### Why is it needed?

Private attributes shouldn't be leaked through the GraphQL schemas, even if they're filtered out anyway.

Also this could cause issues due to required private fields in the inputs.

### How to test it?

- Create a content-type with public and private attributes
- Head to http://localhost:1337/graphql
- Navigate to the schemas inputs and filters types and make sure there is no mention of the private attribute

### Related issue(s)/PR(s)

fix #16325
